packages/app/src/sections/futures/Trade/TransferSmartMarginModal.tsx

```diff
@@ -0,0 +1,271 @@
+import {
+	selectAvailableMargin,
+	selectIsSubmittingCrossTransfer,
+	selectPosition,
+	selectWithdrawableMargin,
+} from 'state/futures/selectors'
+import { useAppDispatch, useAppSelector } from 'state/hooks'
+
+type Props = {
+	onDismiss(): void
+	defaultTab: 'deposit' | 'withdraw'
+}
+
+type BalanceStatus = 'low_balance' | 'no_balance' | 'high_balance'
+
+const SocketBridge = dynamic(() => import('../../../components/SocketBridge'), {
+	ssr: false,
+})
+
+const PLACEHOLDER = '$0.00'
+
+const TransferSmartMarginModal: React.FC<Props> = ({ onDismiss, defaultTab }) => {
+	const { t } = useTranslation()
+	const dispatch = useAppDispatch()
+
+	const position = useAppSelector(selectPosition)
+	const submitting = useAppSelector(selectIsSubmittingCrossTransfer)
+	const totalWithdrawable = useAppSelector(selectWithdrawableMargin)
+	const transactionState = useAppSelector(selectTransaction)
+	const susdBalance = useAppSelector(selectSusdBalance)
+	const availableMargin = useAppSelector(selectAvailableMargin)
+
+	const minDeposit = useMemo(() => {
+		const accessibleMargin = position?.accessibleMargin ?? ZERO_WEI
+		const min = MIN_MARGIN_AMOUNT.sub(accessibleMargin)
+		return min.lt(ZERO_WEI) ? ZERO_WEI : min
+	}, [position?.accessibleMargin])
+
+	const [openSocket, setOpenSocket] = useState(false)
+	const [amount, setAmount] = useState('')
+	const [transferType, setTransferType] = useState(defaultTab === 'deposit' ? 0 : 1)
+
+	const susdBal = transferType === 0 ? susdBalance : totalWithdrawable
+
+	const balanceStatus: BalanceStatus = useMemo(
+		() =>
+			availableMargin.gt(ZERO_WEI) || susdBalance.gt(minDeposit)
+				? 'high_balance'
+				: susdBalance.eq(ZERO_WEI)
+				? 'no_balance'
+				: 'low_balance',
+		[availableMargin, minDeposit, susdBalance]
+	)
+
+	useEffect(() => {
+		switch (balanceStatus) {
+			case 'no_balance':
+			case 'low_balance':
+				setOpenSocket(true)
+				break
+			case 'high_balance':
+				setOpenSocket(false)
+		}
+	}, [balanceStatus])
+
+	const isDisabled = useMemo(() => {
+		if (!amount || submitting) {
+			return true
+		}
+		const amtWei = wei(amount)
+		if (amtWei.eq(0) || amtWei.gt(susdBal) || (transferType === 0 && amtWei.lt(minDeposit))) {
+			return true
+		}
+		return false
+	}, [amount, susdBal, minDeposit, transferType, submitting])
+
+	const computedWithdrawAmount = useMemo(
+		() => (availableMargin.eq(wei(amount || 0)) ? availableMargin : wei(amount || 0)),
+		[amount, availableMargin]
+	)
+
+	const onChangeShowSocket = useCallback(() => setOpenSocket(!openSocket), [openSocket])
+
+	const handleSetMax = useCallback(() => {
+		if (transferType === 0) {
+			setAmount(susdBal.toString())
+		} else {
+			setAmount(totalWithdrawable.toString())
+		}
+	}, [transferType, susdBal, totalWithdrawable])
+
+	const onChangeTab = (selection: number) => {
+		setTransferType(selection)
+		setAmount('')
+	}
+
+	const onDeposit = () => {
+		dispatch(depositCrossMargin(wei(amount)))
```

Comment:
There's no need to deposit to smart margin, this should just be about accessing the socket bridge.

Reply:
Yes. It's unnecessary. Removed.